### PR TITLE
Remove +Q and +e flags from being generated if using R15

### PR DESCRIPTION
**Pull request by jaredmorrow**
Erlang R15 does not support the +Q or +e flags.  This change
simple applies a translation to remove those flags if
R15 is being used.

### CODE REVIEW

**Comment by seancribbs** [comment]
As @joedevivo pointed out to me privately, this fixes the issue but breaks the unit tests for the schema.

**Comment by joedevivo** [comment]
@seancribbs and I worked out a solution today that will be able to tell the difference between R15 and R16 versions of these options (shoutout to Bjorn). Watch this PR tomorrow.

**Comment by jaredmorrow** [comment]
Passed tests on R15B01, R16B02, and R16B02-basho4

+1

File: src/cuttlefish.erl [hunk 1]
```diff
@@ -22,13 +22,59 @@
...
+otp(DesiredMinimumOTPVersion, IfGreaterOrEqual, IfLessThan) ->
+    ActualOTPVersion = erlang:system_info(otp_release),
+    case otp(DesiredMinimumOTPVersion, ActualOTPVersion) of
+        true -> IfGreaterOrEqual;
+        _ -> IfLessThan
+    end.
+
+% @doc is ActualOTPVersion >= DesiredMinimumOTPVersion?
+-spec otp(string(), string()) -> boolean().
+otp(DesiredMinimumOTPVersion, ActualOTPVersion) ->
```
Comment (jaredmorrow) [inline src/cuttlefish.erl]: Only comment I have is the unusual naming of this function.  I know it is test code, but I wouldn't expect otp/2 to return true/false based on one arg being greater than the other.  This is a just a weak comment, I don't feel strongly enough to ask anyone to change it.

File: src/cuttlefish.erl [hunk 2]
```diff
@@ -76,3 +122,14 @@ unset() ->
 -spec invalid(string()) -> no_return().
 invalid(Reason) ->
     throw({invalid, Reason}).
+
+-ifdef(TEST).
+
+otp_test() ->
+    ?assert(otp("R15B01", "R15B02")),
+    ?assert(otp("R15B01", "R15B02-basho3")),
```
Comment (jaredmorrow) [inline src/cuttlefish.erl:125]: I think a more useful test would be to change that to:

``` erlang
?assert(otp("R15B02", "R15B02-basho3")),
```

File: src/cuttlefish.erl [hunk 3]
```diff
@@ -76,3 +122,20 @@ unset() ->
...
+    ?assert(otp("R15B02", "R15B02-basho3")),
+    ?assert(not(otp("R15B02-basho3", "R15B02"))),
+    ?assert(otp("R16B02-basho3", "R16B03")),
+    ?assert(otp("R15B01", "R15B02")),
+    ?assert(otp("R15B01", "R15B02-basho3")),
+    ?assert(not(otp("R16B01", "R15B02"))),
+    ?assert(otp("R16", "R16B03")),
+    ?assert(otp("R16", "R16A")),
+    ?assert(not(otp("R16A", "R16"))),
+    ?assert(not(otp("R16", "R16"))),
```
Comment (jaredmorrow) [inline src/cuttlefish.erl]: this should be `?assert(otp("R16", "R16")),` or you are lying about the whole greater than or 'equal'

Comment (seancribbs) [inline src/cuttlefish.erl]: Agree with @jaredmorrow. Also, the second argument should only be real OTP version strings. There was no "R16", but there was "R16A", "R16B", and so on.

Comment (jaredmorrow) [inline src/cuttlefish.erl]: okay, so change it to `?assert(otp("R16A", "R16A")),`

Comment (joedevivo) [inline src/cuttlefish.erl]: To be fair, I don't claim to check that the version string is valid

**Comment by jaredmorrow** [comment]
Changing the comparison to fact check the claim that equals to be true led to an eunit failure:

```
in function cuttlefish:'-otp_test/0-fun-10-'/0 (src/cuttlefish.erl, line 139)
in call from cuttlefish:otp_test/0 (src/cuttlefish.erl, line 139)
**error:{assertion_failed,[{module,cuttlefish},
                   {line,139},
                   {expression,"otp ( \"R16A\" , \"R16A\" )"},
                   {expected,true},
                   {value,false}]}
```

**Comment by joedevivo** [comment]
I see where to problem is. I'll take care of it when I get back to the computer

**Comment by joedevivo** [comment]
OMG it's so much less complicated now.

**Comment by seancribbs** [comment]
+1 passes on R15B01, R16B01, R16B02-basho3

**Comment by joedevivo** [comment]
LMAO they changed the naming convention and R17 will just be "17.0"

**Comment by jaredmorrow** [comment]
Well we have at least another major before having to worry about it.  So
for now just file an issue.

**Pull request merged**